_posts/2024-04-15-video-segmentation.md

```diff
@@ -0,0 +1,176 @@
+{: refdef}
+
+Note the differences in precision when comparing with the reported results in Youtube-VIS-2021 dataset. This is due to the increase in video sizes, which forces the mode to work in online model, processing chunks of the video that then need to be merged.
+
+<div class="table-wrapper" markdown="block">
+
+| Method | Backbone   | AP                           |
+|--------|------------|------------------------------|
+| IFC    | ResNet-101 | 35.2  (36.6 reported in [4]) |
+| TeViT  | MsgShifT   | **37.9**                     |
+
+</div>
+
+{:refdef: class="image-caption"}
+*Table 2. Comparisons on YouTube-VIS-2021 dataset from TeViT paper [5].*
+{: refdef}
+
+### VisTR (2021)
+
+VisTR, short for VIS Transformer, emerged as one of the initial transformer-based VIS methods to achieve notable accuracy on the YouTube-VIS dataset, thanks to an effective adaptation of DETR [2] for segmentation. This framework processes a fixed sequence of video frames using a ResNet backbone to independently extract features from each image. These extracted features are then concatenated, enriched with a 3D positional encoding, and inputted into an encoder-decoder transformer architecture, which outputs a sequence of object predictions in order.
+
+<div class="post-center-image">
+    {% picture pimage /assets/images/fullsize/posts/2024-04-15-video-segmentation/vistr.png --alt VisTR architecture diagram  %}
+</div>
+
+{:refdef: class="image-caption"}
+*VisTR architecture diagram*
+{: refdef}
+
+Things we need to highlight about this method:
+
+1. Instance queries are fixed, learnable parameters that determine the number of instances that can be predicted (input of decoder in the diagram).
+2. Training involves instance sequence prediction over $N$ frames that requires ground truth matching to compute the loss.
+3. The Hungarian algorithm is used to find a bipartite matching between ground-truth and prediction (as in DETR [2]).
+4. Solving 3 using masks is very expensive. A module is added to obtain bounding boxes from instance predictions and solve the matching efficiently.
+5. The loss consists on a combination of the bounding box, the mask and the class prediction errors.
+6. A 3D conv module fuses temporal information before masks are obtained.
+
+**Drawbacks**: The models are trained on 8 V100 GPUs of 32G RAM, with 1 video clip per GPU. Either low resolution or short clips are used to fit in memory. VisTR remains as a complete offline strategy because it takes the entire video as an input (from IFC paper).
+
+### IFC
+
+Inter-frame Communication Transformers (IFC) leverage the idea that since humans can summarize scenes briefly and consecutive frames often share similarities, it's feasible to communicate frame differences with minimal data. To reduce computational load, IFC utilizes a number of 'memory tokens' to exchange information between frames, thus lowering the complexity of space-time attention.
```

Comment:
```suggestion
Inter-frame Communication Transformers (IFC) leverages the idea that, since humans can summarize scenes briefly and consecutive frames often share similarities, it's feasible to communicate frame differences with minimal data. To reduce computational load, IFC utilizes a number of 'memory tokens' to exchange information between frames, thus lowering the complexity of space-time attention.
```